sql/expression_value.cc

```diff
@@ -3292,15 +3315,42 @@ void
 ExpressionValue::
 mergeToRowDestructive(StructValue & row)
 {
-    row.reserve(row.size() + rowLength());
+    // This is optimized here because the SelectExpression uses it whenever
+    // merging
+    switch (type_) {
+    case Type::STRUCTURED:
+        if (structured_.unique()) {
+            // We have the only reference in the shared pointer, AND we have
```

Comment:
Couldnt there still be many references to a shared ptr of the Expression Value? Would it be safe then?